### ISSUE TITLE
Inductor chooses too large of a block size in cases where the `YBLOCK` dimension is too large.

### ISSUE THREAD

**Issue by Chillee** [pytorch/pytorch#112454]
### 🐛 Describe the bug

```python
import torch

torch.set_default_device('cuda')

@torch.compile
def f(x, y):
    return x.t() + y

f(torch.randn(2**25, 128), torch.randn(128, 2**25))
```

The concrete issue is that this results in us potentially choosing a config like `XBLOCK=256, YBLOCK=512`, which requires too much shared memory.

The reason we end up in this situation is: https://github.com/pytorch/pytorch/blob/main/torch/_inductor/triton_heuristics.py#L810

Basically, because we are limited to launching 65536 blocks on the second/third dim, `triton_config` will elect to scale up `YBLOCK` until we "fit" within the limit.

In this case, we start with a config like `XBLOCK=256, YBLOCK=32`, but we end up scaling `YBLOCK` to 512.

Possible solutions are:
1. Stop launching 2d configs, and just flatten it down to one axis of threadblocks.
2. Choose the XBLOCK axis to be the "large" one.
3. Scale down XBLOCK if `XBLOCK * RBLOCK` is too large.

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

**Comment by shunting314** [comment]
I'll work on this one. cc @Chillee
